src/linux_mcp_server/audit.py

```diff
@@ -90,80 +96,109 @@ def process(self, msg, kwargs):
     yield adapter
 
 
-def log_tool_call(tool_name: str, parameters: dict[str, t.Any]):
-    """
-    Log a tool invocation.
-
-    Args:
-        tool_name: Name of the tool being called
-        parameters: Tool parameters (will be sanitized)
-    """
-    logger = logging.getLogger(__name__)
-
-    # Determine if local or remote execution
-    execution_mode = "remote" if parameters.get("host") else "local"
-
-    # Sanitize parameters
-    safe_params = sanitize_parameters(parameters)
+def _log_event_start(
+    logger: logging.Logger,
+    tool_name: str,
+    execution_mode: str,
+    params: dict[t.Any, t.Any],
+) -> int:
+    execution_mode = "remote" if params.get("host") else "local"
+    safe_params = sanitize_parameters(params)
 
-    # Build log record with extra fields
     extra = {
         "event": "TOOL_CALL",
```

Comment:
I feel like this event value should match the name of the method somehow. `_log_event_tool_call`? Also would it make sense to move these strings into an enum?

Reply:
The full log message includes the name of the tool being called. The actual function doing the calling seems more of an implementation detail more useful to the developer than the user.

```
2025-11-05 16:20:41 | INFO | linux-mcp-server | TOOL_CALL: list_services | event=TOOL_CALL | tool=list_services | execution_mode=local
2025-11-05 16:20:41 | INFO | linux-mcp-server | TOOL_COMPLETE: list_services | event=TOOL_COMPLETE | tool=list_services | status=success | duration=0:00:00.000001s
```

Reply:
I wouldn't mind making these values enums. Currently it's only used in this one file apart from tests.

Reply:
I was thinking that `execution_mode` should be an enum.